src/object/srv_ec_aggregate.c

```diff
@@ -2379,25 +2393,24 @@ agg_iv_ult(void *arg)
 	if (rc)
 		goto out;
 
-	agg_param->ap_prop = daos_prop_alloc(0);
-	if (agg_param->ap_prop == NULL) {
+	prop = daos_prop_alloc(0);
+	if (prop == NULL) {
 		D_ERROR("Property allocation failed\n");
 		rc = -DER_NOMEM;
 		goto out;
 	}
 
-	rc = ds_pool_iv_prop_fetch(agg_param->ap_pool_info.api_pool,
-				   agg_param->ap_prop);
+	rc = ds_pool_iv_prop_fetch(agg_param->ap_pool_info.api_pool, prop);
 	if (rc) {
 		D_ERROR("ds_pool_iv_prop_fetch failed: "DF_RC"\n", DP_RC(rc));
 		goto out;
 	}
 
-	entry = daos_prop_entry_get(agg_param->ap_prop, DAOS_PROP_PO_SVC_LIST);
+	entry = daos_prop_entry_get(prop, DAOS_PROP_PO_SVC_LIST);
 	D_ASSERT(entry != NULL);
-	agg_param->ap_pool_info.api_svc_list =
-		(d_rank_list_t *)entry->dpe_val_ptr;
-
+	d_rank_list_dup(&agg_param->ap_pool_info.api_svc_list,
+			(d_rank_list_t *)entry->dpe_val_ptr);
```

Comment:
possibly fail and get NULL/invalid agg_param->ap_pool_info.api_svc_list?

Reply:
indeed